### DISCUSSION TITLE
登录时“无法访问此页面”的问题解决办法

### DISCUSSION THREAD

**Discussion by JimmyKmi** [XiaoMi/ha_xiaomi_home#6] [Q&A] [answered]
中文:
登录集成时，遇到“无法访问此页面”的提示时，可以看看网址是不是 https://homeassistant.local:8123/api/xxxx。
这是由于你无法通过“homeassistant.local”连接到你的本地HA，可以尝试将“homeassistant.local”改成你的HomeAssistant IP解决。
https://homeassistant.local:8123/api/xxxx -> https://192.168.1.10:8123/api/xxxx ，当然端口号也可以更改

EN:
When encountering the "Unable to access this page" message during the login integration, you might want to check if the URL is https://homeassistant.local:8123/api/xxxx. This issue occurs because you cannot connect to your local Home Assistant using "homeassistant.local." You can try replacing "homeassistant.local" with your Home Assistant IP address to resolve the issue.

eg: https://homeassistant.local:8123/api/xxxx -> https://192.168.1.10:8123/api/xxxx , You can also change the port number if needed.

**Comment by lovemehx** [comment]
如何将“homeassistant.local”改成HomeAssistant IP

**Reply by JimmyKmi** [reply, marked as answer]
比如：https://homeassistant.local:8123/api/xxxx -> https://192.168.1.10:8123/api/xxxx，当然端口号也可以更改

**Reply by lovemehx** [reply]
已经成功了，非常感谢

**Reply by realzzc** [reply]
太感谢了！已经成功~

**Reply by metuly** [reply]
![image](https://github.com/user-attachments/assets/92142c48-0d24-41e8-98d2-f9d6b638b42c)我卡在这个状态

**Reply by renjiaqi100-maker** [reply]
> ![image](https://private-user-images.githubusercontent.com/121495141/516887271-92142c48-0d24-41e8-98d2-f9d6b638b42c.jpeg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njk0MDE0OTcsIm5iZiI6MTc2OTQwMTE5NywicGF0aCI6Ii8xMjE0OTUxNDEvNTE2ODg3MjcxLTkyMTQyYzQ4LTBkMjQtNDFlOC05OGQyLWY5ZDZiNjM4YjQyYy5qcGVnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDEyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjAxMjZUMDQxOTU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGNlOGNiMGQzZjZlZTFjNTg4NWFhYzc0MjhiNGI4ZDExZmI0NzZhYTM1MjkzYjkyNjhmMTdlMGEwMzI3NmY0NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.u9RVeQP13GG_G04lqKVVmWO8YEFLNlc4VZKJvn7cUEY)我卡在这个状态

我也卡在这里了，过不去

**Comment by crazyurus** [comment]
如果改过端口号，点击登录并点击授权，浏览器会卡在 homeassistant.local:8123，此时把浏览器 URL 改成自己的 ip:port 即可完成授权。注意 https 改成 http

**Reply by diametrical** [reply]
没看明白，怎么改呀？

**Reply by crazyurus** [reply]
改卡住时的浏览器 URL，卡住是因为端口是默认值，改成你的实际值

**Reply by lovemehx** [reply]
非常感谢！

**Reply by diametrical** [reply]
哭了，我的URL跳转太快，终于找到了。

**Reply by nforce4-csu** [reply]
感谢，已经修改成功

**Reply by kid1412621** [reply]
> 哭了，我的URL跳转太快，终于找到了。

不要用 Safari，有 bug

**Reply by yeendoin** [reply]
可能之前修改成设备IP地址可以成功，但现在修改IP或是修改host文件都不行。
请问大家有这种情况吗

**Comment by qinqinbaby** [comment]
无法跳转的通过这个方法试试,其实就是在无法跳转的那个画面直接改一下内网的地址就可以了
![1](https://github.com/user-attachments/assets/6ece226e-f6f8-4b58-9830-ca5c15c093a1)
![2](https://github.com/user-attachments/assets/d09a0156-dd9e-463c-b1a4-655d658b7b30)
![3](https://github.com/user-attachments/assets/d6077072-06f7-42b6-aaa7-e4ab8c9b8076)
![4](https://github.com/user-attachments/assets/ec07e703-6d60-48c2-b487-45bdc001d2f5)
![5](https://github.com/user-attachments/assets/3a0de0a2-7e2d-4171-9c0f-bdbfbbacd07a)

**Comment by windfalls07** [comment]
![Screenshot 2024-12-22 225205](https://github.com/user-attachments/assets/349cdcfb-028d-48b9-86d4-cb3bbb6bed98)
我改成自己的ip以后窗口直接消失了，回到原来的页面上他显示 already in progress,然后也没有任何下一步可以操作，这种怎么办

**Reply by hanlinkan** [reply]
我也是这样

**Reply by moliran** [reply]
我是网页消失，显示获取登录授权信息（OAuth token）失败

**Reply by lubia96** [reply]
同问

**Reply by lubia96** [reply]
> 我是网页消失，显示获取登录授权信息（OAuth token）失败

我也是这样，无限重复，你后来成功了吗

**Reply by adl10929** [reply]
我现在也是这情况，有人解决了嘛？

**Comment by yiping520** [comment]
改IP确实可以的

**Comment by ojbk13** [comment]
![20000](https://github.com/user-attachments/assets/70cc396f-c4cd-485b-84a2-3b19f2c43ba3)
扎心了

**Comment by jiyeliu** [comment]
![image](https://github.com/user-attachments/assets/a2e5b04c-7aab-40de-ab5e-c42ff81fed27)
登录之后出现这个怎么解决

**Reply by yifan-lee** [reply]
我也遇到了同样的问题，把homeassistant.local:8123改成ip.8123之后网页直接消失，然后出现这个错误

**Comment by Anzic23** [comment]
Same problem! Is it really impossible to add redirects to custom url???

**Comment by WilliamLuuu** [comment]
直接在Dnsmasq里加一条自定义，把域名挟持到HA所在的IP上就行了

**Comment by imadreamerboy** [comment]
replacing the URL works but u might have to change from https to http depending on setup.
替换 URL 可以奏效，但根据设置不同，可能需要将 https 改为 http。

**Comment by nikhilroycomgmaeroy-sudo** [comment]
Hlo

**Comment by singularity-s0** [comment]
You can add a /etc/hosts entry on your local device, pointing homeassistant.local to your device's IP.

**Comment by MinLonger** [comment]
提供另外一个方法，最新的xiaomi home 好像将端口8123改成默认的80了，所以我直接将验证中的地址中的 “：8123”删除掉，就好了。ip没有替换进去也验证成功了。 供参考

**Reply by 1105162862** [reply]
it works!
